pkg/sfu/videolayerselector/selectordecisioncache.go

```diff
@@ -33,18 +33,23 @@ func (s selectorDecision) String() string {
 // ----------------------------------------------------------------------
 
 type SelectorDecisionCache struct {
-	initialized bool
-	base        uint64
-	last        uint64
-	masks       []uint64
-	numEntries  uint64
+	initialized    bool
+	base           uint64
+	last           uint64
+	masks          []uint64
+	numEntries     uint64
+	numNackEntries uint64
+
+	onExpectEntityChanged map[uint64][]func(entity uint64, decision selectorDecision)
 }
 
-func NewSelectorDecisionCache(maxNumElements uint64) *SelectorDecisionCache {
+func NewSelectorDecisionCache(maxNumElements uint64, numNackEntries uint64) *SelectorDecisionCache {
 	numElements := (maxNumElements*2 + 63) / 64
 	return &SelectorDecisionCache{
-		masks:      make([]uint64, numElements),
-		numEntries: numElements * 32, // 2 bits per entry
+		masks:                 make([]uint64, numElements),
+		numEntries:            numElements * 32, // 2 bits per entry
+		numNackEntries:        numNackEntries,
+		onExpectEntityChanged: make(map[uint64][]func(entity uint64, decision selectorDecision)),
```

Comment:
can this map grow to a large size under very bad conditions?

Reply:
Looks like it may grow up to NACK size? which is 80 here?

Reply:
The map will be clear every time we add an entity to ensure it will not exceed the numEntries.

```
	for e, fns := range s.onExpectEntityChanged {
		if e+s.numEntries < s.last {
			delete(s.onExpectEntityChanged, e)
			for _, f := range fns {
				f(e, selectorDecisionMissing)
			}
		}
	}
```